### ISSUE TITLE
Add RProp in differencial evolution

### ISSUE THREAD

**Issue by EliteFriendly** [EliteFriendly/SelfConfig-GP-for-NeuronNetwork#3]
To increase the speed of neural network training, we can add another algorithm to the fitness function calculation. This algorithm uses derivatives of the loss function, and it doesn't increase the amount of calculations. This algorithm is called Resilient Propagation (Rprop).
